sdk/communication/Azure.Communication.JobRouter/README.md

```diff
@@ -0,0 +1,298 @@
+        Labels = new Dictionary<string, LabelValue>()
+        {
+            ["Some-Skill"] = new LabelValue(11)
+        },
+        ChannelConfigurations = new Dictionary<string, ChannelConfiguration>()
+        {
+            ["my-channel"] = new ChannelConfiguration(1)
+        },
+        AvailableForOffers = true,
+    }
+);
+```
+
+### Offer
+We should get a [RouterWorkerOfferIssued][offer_issued_event_schema] from our [EventGrid subscription][subscribe_events].
+
+There are several different Azure services that act as a [event handler][event_grid_event_handlers].
+For this scenario, we are going to assume Webhooks for event delivery. [Learn more about Webhook event delivery][webhook_event_grid_event_delivery]
+
+Once events are delivered to the event handler, we can deserialize the JSON payload into a list of events.
+
+```C#
+// Parse the JSON payload into a list of events
+EventGridEvent[] egEvents = EventGridEvent.ParseMany(BinaryData.FromStream(httpContent));
+```
+
+```C#
+string offerId = null;
+foreach (EventGridEvent egEvent in egEvents)
+{
+    // If the event is a system event, TryGetSystemEventData will return the deserialized system event
+    if (egEvent.TryGetSystemEventData(out object systemEvent))
+    {
+        switch (systemEvent)
+        {
+            case SubscriptionValidationEventData subscriptionValidated:
+                Console.WriteLine(subscriptionValidated.ValidationCode);
+                break;
+            ...
+            ...
+            case AcsRouterWorkerOfferIssuedEventData issuedOffer:
+                Console.WriteLine($"Received offer with id: {issuedOffer.OfferId}");
+                offerId = issuedOffer.OfferId;
+                break;
+             ...
+             ...
+             ...
+            // Handle any other system event type
+            default:
+                Console.WriteLine(egEvent.EventType);
+                // we can get the raw Json for the event using Data
+                Console.WriteLine(egEvent.Data.ToString());
+                break;
+        }
+    }
+}
+```
+
+However, we could also wait a few seconds and then query the worker directly against the JobRouter API to see if an offer was issued to it.
+```C# Snippet:Azure_Communication_JobRouter_Tests_Samples_QueryWorker_Async
+var result = await routerClient.GetWorkerAsync(worker.Value.Id);
+foreach (var offer in result.Value.Offers)
+{
+    Console.WriteLine($"Worker {worker.Value.Id} has an active offer for job {offer.JobId}");
+}
+```
+
+### Accept an offer
+Once a worker receives an offer, it can take two possible actions: accept or decline. We are going to accept the offer.
+```C# Snippet:Azure_Communication_JobRouter_Tests_Samples_AcceptOffer_Async
+// fetching the offer id
+var jobOffer = result.Value.Offers.FirstOrDefault(x => x.JobId == job.Value.Id);
+
+var offerId = jobOffer!.Id; // `OfferId` can be retrieved directly from consuming event from Event grid
```

Comment:
We generally encourage avoiding nullable reference use in sample snippets.

Reply:
This has been fixed